### PR TITLE
Commit review fixes and fix clippy warnings

### PR DESCRIPTION
**Pull request by F3Joule**

### CODE REVIEW

**Comment by F3Joule** [comment]
[Rename `Change` to `WhoAndWhen` and move it to separate pallet](https://trello.com/c/NITssrMY/457-rename-change-to-whoandwhen-and-move-it-to-separate-pallet)

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -237,18 +237,18 @@ impl pallet_social::Trait for Runtime {
 parameter_types! {
   pub const MinSpaceOwners: u16 = 1;
   pub const MaxSpaceOwners: u16 = u16::max_value();
```
Comment (siman) [inline runtime/src/lib.rs]: I just realized that `65536` space owners could be too much 😅 .
```suggestion
  pub const MaxSpaceOwners: u16 = 1000;
```

File: pallets/space-owners/src/functions.rs [hunk 2]
```diff
@@ -11,52 +11,44 @@ impl<T: Trait> Module<T> {
...
-    <TxById<T>>::insert(tx_id, tx);
-    Self::deposit_event(RawEvent::SpaceOwnersUpdated(who, space_id, tx_id));
+    <ChangeById<T>>::insert(change_id, change);
+    Self::deposit_event(RawEvent::SpaceOwnersUpdated(who, space_id, change_id));
 
     Ok(())
   }
 
-  pub fn change_tx_from_pending_to_executed(space_id: SpaceId, tx_id: TransactionId) -> DispatchResult {
+  pub fn change_change_from_pending_to_executed(space_id: SpaceId, change_id: ChangeId) -> DispatchResult {
```
Comment (siman) [inline pallets/space-owners/src/functions.rs]:
```suggestion
  pub fn move_change_from_pending_state_to_executed(space_id: SpaceId, change_id: ChangeId) -> DispatchResult {
```

File: pallets/space-owners/src/functions.rs [hunk 3]
```diff
@@ -11,52 +11,44 @@ impl<T: Trait> Module<T> {
...
-  pub fn update_space_owners(who: T::AccountId, mut space_owners: SpaceOwners<T>, tx: Transaction<T>) -> DispatchResult {
+  pub fn update_space_owners(who: T::AccountId, mut space_owners: SpaceOwners<T>, change: Change<T>) -> DispatchResult {
     let space_id = space_owners.space_id;
-    let tx_id = tx.id;
+    let change_id = change.id;
 
-    ensure!(tx.confirmed_by.len() >= space_owners.threshold as usize, Error::<T>::NotEnoughConfirms);
-    Self::change_tx_from_pending_to_executed(space_id, tx_id)?;
+    ensure!(change.confirmed_by.len() >= space_owners.threshold as usize, Error::<T>::NotEnoughConfirms);
+    Self::change_change_from_pending_to_executed(space_id, change_id)?;
```
Comment (siman) [inline pallets/space-owners/src/functions.rs]:
```suggestion
    Self::move_change_from_pending_state_to_executed(space_id, change_id)?;
```

File: pallets/space-owners/src/functions.rs [hunk 4]
```diff
@@ -77,14 +69,14 @@ impl<T: Trait> Module<T> {
...
-  pub fn clean_pending_txs(block_number: T::BlockNumber) {
-    if (block_number % T::CleanExpiredTxsPeriod::get()).is_zero() {
-      for tx_id in Self::pending_tx_ids() {
-        if let Some(tx) = Self::tx_by_id(tx_id) {
-          if block_number >= tx.expires_at {
-            PendingTxIdBySpaceId::remove(&tx.space_id);
-            <TxById<T>>::remove(&tx_id);
-            PendingTxIds::mutate(|set| set.remove(&tx_id));
+  pub fn delete_expired_changes(block_number: T::BlockNumber) {
+    if (block_number % T::CleanExpiredChangesPeriod::get()).is_zero() {
```
Comment (siman) [inline pallets/space-owners/src/functions.rs]:
```suggestion
    if (block_number % T::DeleteExpiredChangesPeriod::get()).is_zero() {
```

File: pallets/space-owners/src/lib.rs [hunk 5]
```diff
@@ -55,22 +48,22 @@ pub trait Trait: system::Trait + pallet_timestamp::Trait {
...
+  /// Maximum change notes length.
+  type MaxChangeNotesLength: Get<u16>;
 
   /// Expiration time for change proposal.
   type BlocksToLive: Get<Self::BlockNumber>;
 
-  /// Period in blocks to initialize cleaning of pending txs that are outdated.
-  type CleanExpiredTxsPeriod: Get<Self::BlockNumber>;
+  /// Period in blocks to initialize cleaning of pending changes that are outdated.
+  type CleanExpiredChangesPeriod: Get<Self::BlockNumber>;
```
Comment (siman) [inline pallets/space-owners/src/lib.rs]:
```suggestion
  type DeleteExpiredChangesPeriod: Get<Self::BlockNumber>;
```

File: pallets/space-owners/src/lib.rs [hunk 6]
```diff
@@ -140,20 +133,20 @@ decl_module! {
...
+    /// Maximum change notes length.
+    const MaxChangeNotesLength: u16 = T::MaxChangeNotesLength::get();
 
     /// Period for which change proposal is active.
     const BlocksToLive: T::BlockNumber = T::BlocksToLive::get();
 
-    /// Period in blocks to initialize cleaning of pending txs that are outdated.
-    const CleanExpiredTxsPeriod: T::BlockNumber = T::CleanExpiredTxsPeriod::get();
+    /// Period in blocks to initialize cleaning of pending changes that are outdated.
+    const CleanExpiredChangesPeriod: T::BlockNumber = T::CleanExpiredChangesPeriod::get();
```
Comment (siman) [inline pallets/space-owners/src/lib.rs]:
```suggestion
    const DeleteExpiredChangesPeriod: T::BlockNumber = T::DeleteExpiredChangesPeriod::get();
```

File: pallets/space-owners/src/lib.rs [hunk 7]
```diff
@@ -140,20 +133,20 @@ decl_module! {
...
-    const MaxTxNotesLength: u16 = T::MaxTxNotesLength::get();
+    /// Maximum change notes length.
+    const MaxChangeNotesLength: u16 = T::MaxChangeNotesLength::get();
 
     /// Period for which change proposal is active.
     const BlocksToLive: T::BlockNumber = T::BlocksToLive::get();
 
-    /// Period in blocks to initialize cleaning of pending txs that are outdated.
-    const CleanExpiredTxsPeriod: T::BlockNumber = T::CleanExpiredTxsPeriod::get();
+    /// Period in blocks to initialize cleaning of pending changes that are outdated.
```
Comment (siman) [inline pallets/space-owners/src/lib.rs]:
```suggestion
    /// Period in blocks to initialize deleting of pending changes that are outdated.
```

File: pallets/space-owners/src/lib.rs [hunk 8]
```diff
@@ -140,20 +133,20 @@ decl_module! {
     /// Maximum space owners allowed.
     const MaxSpaceOwners: u16 = T::MaxSpaceOwners::get();
 
-    /// Maximum transaction notes length.
-    const MaxTxNotesLength: u16 = T::MaxTxNotesLength::get();
+    /// Maximum change notes length.
+    const MaxChangeNotesLength: u16 = T::MaxChangeNotesLength::get();
 
     /// Period for which change proposal is active.
```
Comment (siman) [inline pallets/space-owners/src/lib.rs]:
```suggestion
    /// Period in blocks for which change proposal is can remain in a pending state until deleted.
```

File: pallets/space-owners/src/tests.rs [hunk 9]
```diff
@@ -54,18 +54,18 @@ impl pallet_timestamp::Trait for Test {
 parameter_types! {
 	pub const MinSpaceOwners: u16 = 1;
 	pub const MaxSpaceOwners: u16 = u16::max_value();
-	pub const MaxTxNotesLength: u16 = 1024;
+	pub const MaxChangeNotesLength: u16 = 1024;
 	pub const BlocksToLive: u64 = 302_400;
-	pub const CleanExpiredTxsPeriod: u64 = 1800;
+	pub const CleanExpiredChangesPeriod: u64 = 1800;
```
Comment (siman) [inline pallets/space-owners/src/tests.rs]:
```suggestion
	pub const DeleteExpiredChangesPeriod: u64 = 1800;
```

File: pallets/space-owners/src/tests.rs [hunk 10]
```diff
@@ -54,18 +54,18 @@ impl pallet_timestamp::Trait for Test {
...
 
 impl Trait for Test {
   type Event = ();
   type MinSpaceOwners = MinSpaceOwners;
   type MaxSpaceOwners = MaxSpaceOwners;
-  type MaxTxNotesLength = MaxTxNotesLength;
+  type MaxChangeNotesLength = MaxChangeNotesLength;
   type BlocksToLive = BlocksToLive;
-  type CleanExpiredTxsPeriod = CleanExpiredTxsPeriod;
+  type CleanExpiredChangesPeriod = CleanExpiredChangesPeriod;
```
Comment (siman) [inline pallets/space-owners/src/tests.rs]:
```suggestion
  type DeleteExpiredChangesPeriod = DeleteExpiredChangesPeriod;
```

File: pallets/space-owners/src/lib.rs [hunk 11]
```diff
@@ -55,22 +48,22 @@ pub trait Trait: system::Trait + pallet_timestamp::Trait {
   /// Maximum space owners allowed.
   type MaxSpaceOwners: Get<u16>;
 
-  /// Maximum transaction notes length.
-  type MaxTxNotesLength: Get<u16>;
+  /// Maximum change notes length.
```
Comment (siman) [inline pallets/space-owners/src/lib.rs]:
```suggestion
  /// Maximum length of change notes.
```

File: pallets/space-owners/src/lib.rs [hunk 12]
```diff
@@ -254,70 +247,70 @@ decl_module! {
...
       } else {
-        <TxById<T>>::insert(tx_id, tx);
+        <ChangeById<T>>::insert(change_id, change);
       }
 
-      Self::deposit_event(RawEvent::ChangeConfirmed(who, space_id, tx_id));
+      Self::deposit_event(RawEvent::ChangeConfirmed(who, space_id, change_id));
     }
 
     pub fn cancel_proposal(
```
Comment (siman) [inline pallets/space-owners/src/lib.rs]:
```suggestion
    pub fn cancel_change(
```

File: pallets/space-owners/src/lib.rs [hunk 13]
```diff
@@ -55,22 +48,22 @@ pub trait Trait: system::Trait + pallet_timestamp::Trait {
...
+  type CleanExpiredChangesPeriod: Get<Self::BlockNumber>;
 }
 
 decl_error! {
   pub enum Error for Module<T: Trait> {
     /// Space owners was not found by id
     SpaceOwnersNotFound,
-    /// Transaction was not found in a space owners
-    TxNotFound,
+    /// Change was not found in a space owners
```
Comment (siman) [inline pallets/space-owners/src/lib.rs]:
```suggestion
    /// Change was not found by id
```

File: pallets/space-owners/src/lib.rs [hunk 14]
```diff
@@ -85,49 +78,49 @@ decl_error! {
...
     /// The required confirmation count can not be greater than owners count"
     TooBigThreshold,
-    /// Transaction notes are too long
-    TxNotesOversize,
-    /// No space owners will left in result of tx
+    /// Change notes are too long
+    ChangeNotesOversize,
+    /// No space owners will left in result of change
     NoSpaceOwnersLeft,
     /// No updates proposed in change proposal
```
Comment (siman) [inline pallets/space-owners/src/lib.rs]:
```suggestion
    /// No updates proposed with this change
```

File: pallets/space-owners/src/lib.rs [hunk 15]
```diff
@@ -85,49 +78,49 @@ decl_error! {
...
     NoUpdatesProposed,
     /// No fields update in result of change proposal
     NoFieldsUpdatedOnProposal,
 
-    /// Account has already confirmed this transaction
-    TxAlreadyConfirmed,
-    /// There are not enough confirmations on a transaction
+    /// Account has already confirmed this change
+    ChangeAlreadyConfirmed,
+    /// There are not enough confirmations on a change
```
Comment (siman) [inline pallets/space-owners/src/lib.rs]:
```suggestion
    /// There are not enough confirmations for this change
```

File: pallets/space-owners/src/lib.rs [hunk 16]
```diff
@@ -85,49 +78,49 @@ decl_error! {
...
-    TxAlreadyExecuted,
-    /// Transaction is not tied to an owed wallet
-    TxNotRelatedToSpace,
-    /// Pending tx already exists
-    PendingTxAlreadyExists,
-    /// Pending tx doesn't exist
-    PendingTxDoesNotExist,
+    /// Change is already executed
+    ChangeAlreadyExecuted,
+    /// Change is not tied to an owed wallet
```
Comment (siman) [inline pallets/space-owners/src/lib.rs]:
```suggestion
    /// Change is not related to this space
```

File: pallets/space-owners/src/lib.rs [hunk 17]
```diff
@@ -85,49 +78,49 @@ decl_error! {
...
+    ChangeNotRelatedToSpace,
+    /// Pending change already exists
+    PendingChangeAlreadyExists,
+    /// Pending change doesn't exist
+    PendingChangeDoesNotExist,
 
     /// Account is not a proposal creator
     NotAProposalCreator,
```
Comment (siman) [inline pallets/space-owners/src/lib.rs]:
```suggestion
    NotAChangeCreator,
```

File: pallets/space-owners/src/lib.rs [hunk 18]
```diff
@@ -85,49 +78,49 @@ decl_error! {
...
 
-    /// Overflow in Wallet executed tx counter when executing tx
-    OverflowExecutingTx,
-    /// Underflow in Wallet pending tx counter when executing tx
-    UnderflowExecutingTx,
+    /// Overflow in Wallet executed change counter
```
Comment (siman) [inline pallets/space-owners/src/lib.rs]:
```suggestion
    /// Overflow when incrementing a counter of executed changes
```

File: pallets/space-owners/src/lib.rs [hunk 19]
```diff
@@ -85,49 +78,49 @@ decl_error! {
...
+    OverflowExecutingChange,
+    /// Underflow in Wallet pending change counter
```
Comment (siman) [inline pallets/space-owners/src/lib.rs]:
```suggestion
    /// Underflow when decrementing a counter of executed changes
```

File: pallets/space-owners/src/lib.rs [hunk 20]
```diff
@@ -140,20 +133,20 @@ decl_module! {
     /// Maximum space owners allowed.
     const MaxSpaceOwners: u16 = T::MaxSpaceOwners::get();
 
-    /// Maximum transaction notes length.
-    const MaxTxNotesLength: u16 = T::MaxTxNotesLength::get();
+    /// Maximum change notes length.
```
Comment (siman) [inline pallets/space-owners/src/lib.rs]:
```suggestion
    /// Maximum length of change notes.
```

File: pallets/space-owners/src/lib.rs [hunk 21]
```diff
@@ -55,22 +48,22 @@ pub trait Trait: system::Trait + pallet_timestamp::Trait {
...
-  type MaxTxNotesLength: Get<u16>;
+  /// Maximum change notes length.
+  type MaxChangeNotesLength: Get<u16>;
 
   /// Expiration time for change proposal.
   type BlocksToLive: Get<Self::BlockNumber>;
 
-  /// Period in blocks to initialize cleaning of pending txs that are outdated.
-  type CleanExpiredTxsPeriod: Get<Self::BlockNumber>;
+  /// Period in blocks to initialize cleaning of pending changes that are outdated.
```
Comment (siman) [inline pallets/space-owners/src/lib.rs]:
```suggestion
  /// Period in blocks for which change proposal is can remain in a pending state until deleted.
```

**Review by siman — CHANGES_REQUESTED** [review]

File: pallets/space-owners/src/tests.rs [hunk 22]
```diff
@@ -54,18 +54,18 @@ impl pallet_timestamp::Trait for Test {
 parameter_types! {
 	pub const MinSpaceOwners: u16 = 1;
 	pub const MaxSpaceOwners: u16 = u16::max_value();
```
Comment (siman) [inline pallets/space-owners/src/tests.rs]:
```suggestion
	pub const MaxSpaceOwners: u16 = 1000;
```

**Review by siman — APPROVED** [review]

**Pull request merged**